model/search/ResultsWatcher.php

```diff
@@ -86,22 +91,32 @@ protected function addIndex(DeliveryExecutionInterface $deliveryExecution)
         $report = \common_report_Report::createSuccess();
         $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
         if ($searchService->supportCustomIndex()) {
-            $id = $deliveryExecution->getIdentifier();
+            $deliveryExecutionId = $deliveryExecution->getIdentifier();
             $user = UserHelper::getUser($deliveryExecution->getUserIdentifier());
             $customFieldService = $this->getServiceLocator()->get(ResultCustomFieldsService::SERVICE_ID);
             $customBody = $customFieldService->getCustomFields($deliveryExecution);
-            $userName = UserHelper::getUserName($user, true);
+
             $body = [
                 'label' => $deliveryExecution->getLabel(),
                 self::INDEX_DELIVERY => $deliveryExecution->getDelivery()->getUri(),
                 'type' => ResultService::DELIVERY_RESULT_CLASS_URI,
                 self::INDEX_TEST_TAKER => $user->getIdentifier(),
-                self::INDEX_TEST_TAKER_NAME => $userName,
-                self::INDEX_DELIVERY_EXECUTION => $id,
+                self::INDEX_TEST_TAKER_FIRST_NAME => UserHelper::getUserFirstName($user, true),
+                self::INDEX_TEST_TAKER_LAST_NAME => UserHelper::getUserLastName($user, true),
+                self::INDEX_TEST_TAKER_NAME => UserHelper::getUserName($user, true),
+                self::INDEX_DELIVERY_EXECUTION => $deliveryExecutionId,
+                self::INDEX_DELIVERY_EXECUTION_START_TIME =>  $this->transformDateTime(
+                    $deliveryExecution->getStartTime()
+                )
             ];
             $body = array_merge($body, $customBody);
             $queueDispatcher = $this->getServiceLocator()->get(QueueDispatcherInterface::SERVICE_ID);
-            $queueDispatcher->createTask(new AddSearchIndexFromArray(), [$id, $body], __('Adding/Updating search index for %s', $deliveryExecution->getLabel()));
+
+            $queueDispatcher->createTask(
+                new AddSearchIndexFromArray(),
+                [$deliveryExecutionId, $body],
+                __('Adding/Updating search index for %s', $deliveryExecution->getLabel())
+            );
         }
         return $report;
```

Comment:
May be that is a different issue, but should we return something other then from https://github.com/oat-sa/extension-tao-outcomeui/pull/357/files#diff-aa0ad26502a0ff7e10712de744e85487352f350326348e7c73ebc616080bdafaR91 ( alway successful report ) , but based on  the result of execution of https://github.com/oat-sa/extension-tao-outcomeui/pull/357/files#diff-aa0ad26502a0ff7e10712de744e85487352f350326348e7c73ebc616080bdafaR115 ?

Reply:
I did not change business logic here. I just refactor so it is easy to read this part.